### DISCUSSION TITLE
Pre announce future certificate

### DISCUSSION THREAD

**Discussion by ITM-Colin** [osm-search/Nominatim#2690] [General]
Hello,

is there any way to get a future certificate before it is configured on "nominatim.openstreetmap.org"?

we are using the Nominatim API in our Project. The Problem is our security department want us to secure the request by checking the "nominatim.openstreetmap.org" certificate of the response against a predefined "nominatim.openstreetmap.org" certificate.
The problem now occurs when the certificate is changed at "nominatim.openstreetmap.org" we can not send any request until the Firewall Team changed the predefined cerificate.
A solution would be to get the future certificate before it is configured on "nominatim.openstreetmap.org" and set this certificate and the old one as allowed. Therefore the question is there any way to request the future certificate of "nominatim.openstreetmap.org"?

Many thanks.

Yours Sincerly
Colin Wübbena

**Comment by otbutz** [comment]
Why don't you simply pin the CA?

**Reply by ITM-Colin** [reply]
Unfortunately our security department forced us to pin the server certificate.

**Comment by mtmail** [comment]
The public service at nominatim.openstreetmap.org uses lets-encrypt SSL certificates which change about every 90 days. https://operations.osmfoundation.org/policies/nominatim/ 

For this level of security requirement you might want to install your instance of Nominatim software behind the firewall. https://nominatim.org/

**Reply by ITM-Colin** [reply]
Thanks for the alternative idea with the self hosted instance
